staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -404,6 +404,11 @@ redirectLoop:
 		if err != nil {
 			return nil, nil, fmt.Errorf("malformed Location header: %v", err)
 		}
+
+		// Only allow redirects to the same host.
+		if location.Host != originalLocation.Host {
```

Comment:
compare scheme as well?

Reply:
I considered that, but I think at least http -> https should be allowed. And if an attacker controls the endpoint enough to send an https redirect response, then they don't gain anything from a downgrade attack.

Reply:
Makes sense